### PR TITLE
fix(jit): AVX to SSE downgrade for scalar mode

### PR DESCRIPTION
**Pull request by eugenels**
Downgrade all avx instructions to sse in scalar mode.
Fixes: #2161

### CODE REVIEW

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/main/c/share/jit/impl/x86.h [hunk 1]
```diff
@@ -74,20 +74,20 @@ namespace questdb::x86 {
 
         c.cmp(rhs, INT_NULL);
         c.je(l_null);
-        c.vcvtsi2ss(r, r, rhs);
+        c.cvtsi2ss(r, rhs);
         c.jmp(l_exit);
         c.bind(l_null);
-        c.vmovss(r, NaN);
```
Comment (puzpuzpuz) [inline core/src/main/c/share/jit/impl/x86.h:80]: Looks like we still have `c.vmovss` and `c.vmovsd` calls in `mem2reg` function. Is it on purpose?

Comment (eugenels) [reply core/src/main/c/share/jit/impl/x86.h:80]: hmm, should be fixed in the first commit

Comment (puzpuzpuz) [reply core/src/main/c/share/jit/impl/x86.h:80]: Sorry, my bad. I wasn't fully awaken when reading the changes.

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
this PR requires at least 5 binaries @eugenels

**Comment by bluestreak01** [comment]
how do we test this ?

**Comment by eugenels** [comment]
> this PR requires at least 5 binaries @eugenels
only for amd64 platforms (we don't have jit for arm)

**Pull request merged**